### PR TITLE
Implement rotary event

### PR DESCRIPTION
**Pull request by JongHeonChoi**

### CODE REVIEW

File: packaging/crosswalk-tizen.spec [hunk 1]
```diff
@@ -56,7 +56,8 @@ cp %{SOURCE1001} .
 
 %build
 export GYP_GENERATORS='ninja'
-GYP_OPTIONS="--depth=."
+GYP_OPTIONS="--depth=.
+-Dextension_host_os=%{profile}"
```
Comment (WonyoungChoi) [inline packaging/crosswalk-tizen.spec]: Change the variable "extension_host_os" to "profile"

File: runtime/browser/web_application.cc [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #include <memory>
 #include <sstream>
 #include <vector>
+#include <efl_extension.h>
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc]: Fix order of including headers. 
https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

File: runtime/browser/web_application.cc [hunk 3]
```diff
@@ -654,6 +655,30 @@ void WebApplication::OnLowMemory() {
...
+    << "document.dispatchEvent(__event);\n"
+    << "\n"
+    << "for (var i=0; i < window.frames.length; i++)\n"
+    << "{ window.frames[i].document.dispatchEvent(__event); }"
+    << "})()";
+  std::string kRotaryEventScript = script.str();
+  if (view_stack_.size() > 0 && view_stack_.front() != NULL)
+    view_stack_.front()->EvalJavascript(kRotaryEventScript.c_str());
+}
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.cc]: Two spaces between the code and the comment.

File: runtime/browser/web_application.h [hunk 4]
```diff
@@ -21,6 +21,7 @@
 #include <list>
 #include <memory>
 #include <string>
+#include <efl_extension.h>
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.h]: fix including order

File: runtime/browser/web_application.h [hunk 5]
```diff
@@ -89,6 +90,10 @@ class WebApplication : public WebView::EventListener {
   virtual void OnUsermediaPermissionRequest(
       WebView* view, const std::string& url,
       std::function<void(bool)> result_handler);
+#ifdef PROFILE_WEARABLE
+  virtual void OnRotaryEvent(WebView* view,
+      Eext_Rotary_Event_Info* info);
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_application.h]: 2 spaces between code and comment

File: runtime/runtime.gyp [hunk 6]
```diff
@@ -56,6 +56,11 @@
           'launchpad',
         ],
       },
+      'conditions': [
+        ['extension_host_os == "wearable"', {
```
Comment (WonyoungChoi) [inline runtime/runtime.gyp]: change extension_host_os to profile

File: runtime/browser/web_view.h [hunk 7]
```diff
@@ -87,6 +88,11 @@ class WebView {
         WebView* /*view*/,
         const std::string& /*url*/,
         std::function<void(bool)> /*result_handler*/) {}
+#ifdef PROFILE_WEARABLE
+    virtual void OnRotaryEvent(
+        WebView* /*view*/,
+        Eext_Rotary_Event_Info* info) {}
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_view.h]: 2 spaces

File: runtime/browser/web_view_impl.cc [hunk 8]
```diff
@@ -199,6 +199,9 @@ void WebViewImpl::Initialize() {
   InitCertificateAllowCallback();
   InitPopupWaitCallback();
   InitUsermediaCallback();
+#ifdef PROFILE_WEARABLE
+  InitRotaryEventCallback();
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_view_impl.cc]: 2 spaces

File: runtime/browser/web_view_impl.cc [hunk 9]
```diff
@@ -825,6 +828,23 @@ void WebViewImpl::InitUsermediaCallback() {
...
+    Eext_Rotary_Event_Info* rotary = event_info;
+    self->listener_->OnRotaryEvent(self->view_, rotary);
+    return EINA_TRUE;
+  };
+
+  // add callback to handle rotary event
+  eext_rotary_object_event_callback_add(ewk_view_, rotary_callback, this);
+  eext_rotary_object_event_activated_set(ewk_view_, EINA_TRUE);
+}
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_view_impl.cc]: 2 spaces

File: runtime/browser/web_view_impl.h [hunk 10]
```diff
@@ -75,6 +75,9 @@ class WebViewImpl {
   void InitCertificateAllowCallback();
   void InitPopupWaitCallback();
   void InitUsermediaCallback();
+#ifdef PROFILE_WEARABLE
+  void InitRotaryEventCallback();
+#endif // PROFILE_WEARABLE
```
Comment (WonyoungChoi) [inline runtime/browser/web_view_impl.h]: 2 spaces

File: runtime/browser/web_view.h [hunk 11]
```diff
@@ -87,6 +88,11 @@ class WebView {
         WebView* /*view*/,
         const std::string& /*url*/,
         std::function<void(bool)> /*result_handler*/) {}
+#ifdef PROFILE_WEARABLE
+    virtual void OnRotaryEvent(
+        WebView* /*view*/,
+        Eext_Rotary_Event_Info* info) {}
```
Comment (myroot) [inline runtime/browser/web_view.h]: Eext_Rotary_Event_Info
Change to newly defined type. 
To remove Dependency of EFL-Extension

**Comment by myroot** [comment]
LGTM

**Pull request merged**